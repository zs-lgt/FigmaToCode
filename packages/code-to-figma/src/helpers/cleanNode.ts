export const cleanNode = node => {
    if (!node) {
        return;
    }
    const { children, parent, ref, innerRef, ...other } = node;
    return other;
};
