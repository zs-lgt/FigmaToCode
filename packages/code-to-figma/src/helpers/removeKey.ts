export const removeKey = node => {
    if (node && node.key) {
        delete node.key;
    }
    if (node && node.children) {
        node.children.forEach(removeKey);
    }
    return node;
};
