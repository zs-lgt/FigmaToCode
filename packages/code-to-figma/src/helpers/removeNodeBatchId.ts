export const removeNodeBatchId = node => {
    if (node && node.pluginData) {
        delete node.pluginData.nodeBatchId;
    }
    if (node && node.children) {
        node.children.forEach(removeNodeBatchId);
    }
    return node;
};
