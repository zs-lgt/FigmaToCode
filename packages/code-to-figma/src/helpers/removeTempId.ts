export const removeTempId = node => {
    if (node && node.pluginData) {
        delete node.pluginData.reactId;
    }
    if (node && node.children) {
        node.children.forEach(removeTempId);
    }
    return node;
};
