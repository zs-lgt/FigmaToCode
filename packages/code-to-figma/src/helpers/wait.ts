import { act } from 'react-test-renderer';

export function wait(amount = 0) {
    return new Promise(resolve => setTimeout(resolve, amount));
}

// Use this in your test after mounting if you need just need to let the query finish without updating the wrapper
export async function actWait(amount = 0) {
    await act(async () => {
        await wait(amount);
    });
}
