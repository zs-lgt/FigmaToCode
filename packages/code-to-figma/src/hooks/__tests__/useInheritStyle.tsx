import { createFigma } from 'figma-api-stub';
import { render } from '../../renderer';
import { wait } from '../../helpers/wait';
import * as React from 'react';
import { Component, Page, Text, View } from '../..';
import { removeNodeBatchId } from '../../helpers/removeNodeBatchId';
import { removeTempId } from '../../helpers/removeTempId';
import { removeKey } from '../../helpers/removeKey';

const removeMeta = node => {
    return removeKey(removeNodeBatchId(removeTempId(node)));
};

describe('useInheritStyle', () => {
    beforeEach(() => {
        // @ts-ignore
        global.figma = createFigma({
            simulateErrors: true
        });
        process.env.REACT_FIGMA_STYLE_INHERITANCE_ENABLED = '1';
    });

    afterEach(() => {
        delete process.env.REACT_FIGMA_STYLE_INHERITANCE_ENABLED;
    });

    it('styles propagating works', async () => {
        await render(
            <Page name="New page" isCurrent>
                <View style={{ fontSize: 50, fontWeight: 'bold', fontFamily: 'Roboto' } as any}>
                    <View style={{ fontSize: 48 } as any}>
                        <View style={{ width: 200, height: 100, backgroundColor: '#dd55aa' }} />
                        <Text style={{ color: '#ffffff' }}>text</Text>
                    </View>
                </View>
            </Page>
        );
        await wait();

        expect(removeMeta(figma.currentPage)).toMatchSnapshot();
    });

    it('styles propagating works over component without useInheritStyle', async () => {
        await render(
            <Page name="New page" isCurrent>
                <View style={{ fontSize: 50, fontWeight: 'bold', fontFamily: 'Roboto' } as any}>
                    <Component>
                        <View style={{ width: 200, height: 100, backgroundColor: '#dd55aa' }} />
                        <Text style={{ color: '#ffffff' }}>text</Text>
                    </Component>
                </View>
            </Page>
        );
        await wait();

        expect(removeMeta(figma.currentPage)).toMatchSnapshot();
    });

    it('styles propagating works with wrapping to component', async () => {
        const Component = () => {
            return (
                <View style={{ fontSize: 48 } as any}>
                    <View style={{ width: 200, height: 100, backgroundColor: '#dd55aa' }} />
                    <Text style={{ color: '#ffffff' }}>text</Text>
                </View>
            );
        };
        await render(
            <Page name="New page" isCurrent>
                <View style={{ fontSize: 50, fontWeight: 'bold', fontFamily: 'Roboto' } as any}>
                    <Component />
                </View>
            </Page>
        );
        await wait();

        expect(removeMeta(figma.currentPage)).toMatchSnapshot();
    });

    it('havent propagating without REACT_FIGMA_STYLE_INHERITANCE_ENABLED flag', async () => {
        delete process.env.REACT_FIGMA_STYLE_INHERITANCE_ENABLED;
        await render(
            <Page name="New page" isCurrent>
                <View style={{ fontSize: 50, fontWeight: 'bold', fontFamily: 'Roboto' } as any}>
                    <View style={{ fontSize: 48 } as any}>
                        <View style={{ width: 200, height: 100, backgroundColor: '#dd55aa' }} />
                        <Text style={{ color: '#ffffff' }}>text</Text>
                    </View>
                </View>
            </Page>
        );
        await wait();

        expect(removeMeta(figma.currentPage)).toMatchSnapshot();
    });
});
