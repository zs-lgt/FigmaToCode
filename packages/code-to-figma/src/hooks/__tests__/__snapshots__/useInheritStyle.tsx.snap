// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useInheritStyle havent propagating without REACT_FIGMA_STYLE_INHERITANCE_ENABLED flag 1`] = `
ExportMixinStub {
  "_backgrounds": Array [
    Object {
      "blendMode": "NORMAL",
      "color": Object {
        "b": 0.9607843160629272,
        "g": 0.9607843160629272,
        "r": 0.9607843160629272,
      },
      "opacity": 1,
      "type": "SOLID",
      "visible": true,
    },
  ],
  "children": Array [
    GeometryMixinStub {
      "_fills": Array [],
      "backgrounds": Array [
        Object {
          "color": Object {
            "b": 0,
            "g": 0,
            "r": 0,
          },
          "opacity": 0,
          "type": "SOLID",
        },
      ],
      "blendMode": "NORMAL",
      "children": Array [
        GeometryMixinStub {
          "_fills": Array [],
          "backgrounds": Array [
            Object {
              "color": Object {
                "b": 0,
                "g": 0,
                "r": 0,
              },
              "opacity": 0,
              "type": "SOLID",
            },
          ],
          "blendMode": "NORMAL",
          "children": Array [
            GeometryMixinStub {
              "_fills": Array [
                Object {
                  "color": Object {
                    "b": 0.6666666666666666,
                    "g": 0.3333333333333333,
                    "r": 0.8666666666666667,
                  },
                  "opacity": 1,
                  "type": "SOLID",
                },
              ],
              "blendMode": "NORMAL",
              "config": Object {
                "isWithoutTimeout": false,
                "simulateErrors": true,
              },
              "constraints": Object {
                "horizontal": "MIN",
                "vertical": "MIN",
              },
              "cornerRadius": 0,
              "cornerSmoothing": 0,
              "effects": Array [],
              "exportSettings": Array [],
              "height": 100,
              "id": "1:2",
              "isMask": false,
              "layoutAlign": "INHERIT",
              "layoutGrow": 0,
              "locked": false,
              "opacity": 1,
              "parent": [Circular],
              "pluginData": Object {
                "isReactFigmaNode": "true",
                "reactStyle": "{\\"width\\":200,\\"height\\":100,\\"backgroundColor\\":\\"#dd55aa\\"}",
              },
              "strokeAlign": "INSIDE",
              "strokeCap": "NONE",
              "strokeWeight": 0,
              "strokes": Array [],
              "type": "RECTANGLE",
              "visible": true,
              "width": 200,
            },
            GeometryMixinStub {
              "_characters": "text",
              "_fills": Array [
                Object {
                  "color": Object {
                    "b": 1,
                    "g": 1,
                    "r": 1,
                  },
                  "opacity": 1,
                  "type": "SOLID",
                },
              ],
              "_fontName": Object {
                "family": "Inter",
                "style": "Regular",
              },
              "_textAutoResize": "WIDTH_AND_HEIGHT",
              "autoRename": false,
              "blendMode": "NORMAL",
              "config": Object {
                "isWithoutTimeout": false,
                "simulateErrors": true,
              },
              "constraints": Object {
                "horizontal": "MIN",
                "vertical": "MIN",
              },
              "effects": Array [],
              "exportSettings": Array [],
              "fontSize": 12,
              "hyperlink": null,
              "id": "1:3",
              "isMask": false,
              "layoutAlign": "INHERIT",
              "layoutGrow": 0,
              "letterSpacing": Object {
                "unit": "PIXELS",
                "value": 0,
              },
              "lineHeight": Object {
                "unit": "AUTO",
              },
              "locked": false,
              "opacity": 1,
              "paragraphIndent": 0,
              "paragraphSpacing": 0,
              "parent": [Circular],
              "pluginData": Object {
                "isReactFigmaNode": "true",
                "reactStyle": "{\\"color\\":\\"#ffffff\\"}",
              },
              "strokeAlign": "INSIDE",
              "strokeCap": "NONE",
              "strokeWeight": 0,
              "strokes": Array [],
              "textAlignHorizontal": "LEFT",
              "textAlignVertical": "TOP",
              "textCase": "ORIGINAL",
              "textDecoration": "NONE",
              "type": "TEXT",
              "visible": true,
            },
          ],
          "clipsContent": false,
          "config": Object {
            "isWithoutTimeout": false,
            "simulateErrors": true,
          },
          "constraints": Object {
            "horizontal": "MIN",
            "vertical": "MIN",
          },
          "cornerRadius": 0,
          "cornerSmoothing": 0,
          "counterAxisAlignItems": "MIN",
          "counterAxisSizingMode": "AUTO",
          "effects": Array [],
          "exportSettings": Array [],
          "guides": Array [],
          "id": "1:4",
          "isMask": false,
          "itemSpacing": 0,
          "layoutAlign": "INHERIT",
          "layoutGrids": Array [],
          "layoutGrow": 0,
          "layoutMode": "NONE",
          "locked": false,
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "parent": [Circular],
          "pluginData": Object {
            "isReactFigmaNode": "true",
            "reactStyle": "{\\"backgroundColor\\":\\"transparent\\",\\"fontSize\\":48}",
          },
          "primaryAxisAlignItems": "MIN",
          "primaryAxisSizingMode": "AUTO",
          "strokeAlign": "INSIDE",
          "strokeCap": "NONE",
          "strokeWeight": 0,
          "strokes": Array [],
          "type": "FRAME",
          "visible": true,
        },
      ],
      "clipsContent": false,
      "config": Object {
        "isWithoutTimeout": false,
        "simulateErrors": true,
      },
      "constraints": Object {
        "horizontal": "MIN",
        "vertical": "MIN",
      },
      "cornerRadius": 0,
      "cornerSmoothing": 0,
      "counterAxisAlignItems": "MIN",
      "counterAxisSizingMode": "AUTO",
      "effects": Array [],
      "exportSettings": Array [],
      "guides": Array [],
      "id": "1:5",
      "isMask": false,
      "itemSpacing": 0,
      "layoutAlign": "INHERIT",
      "layoutGrids": Array [],
      "layoutGrow": 0,
      "layoutMode": "NONE",
      "locked": false,
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
      "parent": [Circular],
      "pluginData": Object {
        "isReactFigmaNode": "true",
        "reactStyle": "{\\"backgroundColor\\":\\"transparent\\",\\"fontSize\\":50,\\"fontWeight\\":\\"bold\\",\\"fontFamily\\":\\"Roboto\\"}",
      },
      "primaryAxisAlignItems": "MIN",
      "primaryAxisSizingMode": "AUTO",
      "strokeAlign": "INSIDE",
      "strokeCap": "NONE",
      "strokeWeight": 0,
      "strokes": Array [],
      "type": "FRAME",
      "visible": true,
    },
  ],
  "config": Object {
    "isWithoutTimeout": false,
    "simulateErrors": true,
  },
  "exportSettings": Array [],
  "id": "1:1",
  "name": "New page",
  "parent": ChildrenMixinStub {
    "children": Array [
      ExportMixinStub {
        "children": Array [],
        "config": undefined,
        "id": "0:1",
        "parent": [Circular],
        "type": "PAGE",
      },
      [Circular],
    ],
    "config": undefined,
    "id": "0:0",
    "type": "DOCUMENT",
  },
  "pluginData": Object {
    "isReactFigmaNode": "true",
    "reactStyle": "{}",
  },
  "type": "PAGE",
}
`;

exports[`useInheritStyle styles propagating works 1`] = `
ExportMixinStub {
  "_backgrounds": Array [
    Object {
      "blendMode": "NORMAL",
      "color": Object {
        "b": 0.9607843160629272,
        "g": 0.9607843160629272,
        "r": 0.9607843160629272,
      },
      "opacity": 1,
      "type": "SOLID",
      "visible": true,
    },
  ],
  "children": Array [
    GeometryMixinStub {
      "_fills": Array [],
      "backgrounds": Array [
        Object {
          "color": Object {
            "b": 0,
            "g": 0,
            "r": 0,
          },
          "opacity": 0,
          "type": "SOLID",
        },
      ],
      "blendMode": "NORMAL",
      "children": Array [
        GeometryMixinStub {
          "_fills": Array [],
          "backgrounds": Array [
            Object {
              "color": Object {
                "b": 0,
                "g": 0,
                "r": 0,
              },
              "opacity": 0,
              "type": "SOLID",
            },
          ],
          "blendMode": "NORMAL",
          "children": Array [
            GeometryMixinStub {
              "_fills": Array [
                Object {
                  "color": Object {
                    "b": 0.6666666666666666,
                    "g": 0.3333333333333333,
                    "r": 0.8666666666666667,
                  },
                  "opacity": 1,
                  "type": "SOLID",
                },
              ],
              "blendMode": "NORMAL",
              "config": Object {
                "isWithoutTimeout": false,
                "simulateErrors": true,
              },
              "constraints": Object {
                "horizontal": "MIN",
                "vertical": "MIN",
              },
              "cornerRadius": 0,
              "cornerSmoothing": 0,
              "effects": Array [],
              "exportSettings": Array [],
              "height": 100,
              "id": "1:2",
              "isMask": false,
              "layoutAlign": "INHERIT",
              "layoutGrow": 0,
              "locked": false,
              "opacity": 1,
              "parent": [Circular],
              "pluginData": Object {
                "isReactFigmaNode": "true",
                "reactStyle": "{\\"width\\":200,\\"height\\":100,\\"backgroundColor\\":\\"#dd55aa\\"}",
              },
              "strokeAlign": "INSIDE",
              "strokeCap": "NONE",
              "strokeWeight": 0,
              "strokes": Array [],
              "type": "RECTANGLE",
              "visible": true,
              "width": 200,
            },
            GeometryMixinStub {
              "_characters": "text",
              "_fills": Array [
                Object {
                  "color": Object {
                    "b": 1,
                    "g": 1,
                    "r": 1,
                  },
                  "opacity": 1,
                  "type": "SOLID",
                },
              ],
              "_fontName": Object {
                "family": "Roboto",
                "style": "Bold",
              },
              "_textAutoResize": "WIDTH_AND_HEIGHT",
              "autoRename": false,
              "blendMode": "NORMAL",
              "config": Object {
                "isWithoutTimeout": false,
                "simulateErrors": true,
              },
              "constraints": Object {
                "horizontal": "MIN",
                "vertical": "MIN",
              },
              "effects": Array [],
              "exportSettings": Array [],
              "fontSize": 48,
              "hyperlink": null,
              "id": "1:3",
              "isMask": false,
              "layoutAlign": "INHERIT",
              "layoutGrow": 0,
              "letterSpacing": Object {
                "unit": "PIXELS",
                "value": 0,
              },
              "lineHeight": Object {
                "unit": "AUTO",
              },
              "locked": false,
              "opacity": 1,
              "paragraphIndent": 0,
              "paragraphSpacing": 0,
              "parent": [Circular],
              "pluginData": Object {
                "isReactFigmaNode": "true",
                "reactStyle": "{\\"fontFamily\\":\\"Roboto\\",\\"fontSize\\":48,\\"fontWeight\\":\\"bold\\",\\"color\\":\\"#ffffff\\"}",
              },
              "strokeAlign": "INSIDE",
              "strokeCap": "NONE",
              "strokeWeight": 0,
              "strokes": Array [],
              "textAlignHorizontal": "LEFT",
              "textAlignVertical": "TOP",
              "textCase": "ORIGINAL",
              "textDecoration": "NONE",
              "type": "TEXT",
              "visible": true,
            },
          ],
          "clipsContent": false,
          "config": Object {
            "isWithoutTimeout": false,
            "simulateErrors": true,
          },
          "constraints": Object {
            "horizontal": "MIN",
            "vertical": "MIN",
          },
          "cornerRadius": 0,
          "cornerSmoothing": 0,
          "counterAxisAlignItems": "MIN",
          "counterAxisSizingMode": "AUTO",
          "effects": Array [],
          "exportSettings": Array [],
          "guides": Array [],
          "id": "1:4",
          "isMask": false,
          "itemSpacing": 0,
          "layoutAlign": "INHERIT",
          "layoutGrids": Array [],
          "layoutGrow": 0,
          "layoutMode": "NONE",
          "locked": false,
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "parent": [Circular],
          "pluginData": Object {
            "isReactFigmaNode": "true",
            "reactStyle": "{\\"backgroundColor\\":\\"transparent\\",\\"fontSize\\":48}",
          },
          "primaryAxisAlignItems": "MIN",
          "primaryAxisSizingMode": "AUTO",
          "strokeAlign": "INSIDE",
          "strokeCap": "NONE",
          "strokeWeight": 0,
          "strokes": Array [],
          "type": "FRAME",
          "visible": true,
        },
      ],
      "clipsContent": false,
      "config": Object {
        "isWithoutTimeout": false,
        "simulateErrors": true,
      },
      "constraints": Object {
        "horizontal": "MIN",
        "vertical": "MIN",
      },
      "cornerRadius": 0,
      "cornerSmoothing": 0,
      "counterAxisAlignItems": "MIN",
      "counterAxisSizingMode": "AUTO",
      "effects": Array [],
      "exportSettings": Array [],
      "guides": Array [],
      "id": "1:5",
      "isMask": false,
      "itemSpacing": 0,
      "layoutAlign": "INHERIT",
      "layoutGrids": Array [],
      "layoutGrow": 0,
      "layoutMode": "NONE",
      "locked": false,
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
      "parent": [Circular],
      "pluginData": Object {
        "isReactFigmaNode": "true",
        "reactStyle": "{\\"backgroundColor\\":\\"transparent\\",\\"fontSize\\":50,\\"fontWeight\\":\\"bold\\",\\"fontFamily\\":\\"Roboto\\"}",
      },
      "primaryAxisAlignItems": "MIN",
      "primaryAxisSizingMode": "AUTO",
      "strokeAlign": "INSIDE",
      "strokeCap": "NONE",
      "strokeWeight": 0,
      "strokes": Array [],
      "type": "FRAME",
      "visible": true,
    },
  ],
  "config": Object {
    "isWithoutTimeout": false,
    "simulateErrors": true,
  },
  "exportSettings": Array [],
  "id": "1:1",
  "name": "New page",
  "parent": ChildrenMixinStub {
    "children": Array [
      ExportMixinStub {
        "children": Array [],
        "config": undefined,
        "id": "0:1",
        "parent": [Circular],
        "type": "PAGE",
      },
      [Circular],
    ],
    "config": undefined,
    "id": "0:0",
    "type": "DOCUMENT",
  },
  "pluginData": Object {
    "isReactFigmaNode": "true",
    "reactStyle": "{}",
  },
  "type": "PAGE",
}
`;

exports[`useInheritStyle styles propagating works over component without useInheritStyle 1`] = `
ExportMixinStub {
  "_backgrounds": Array [
    Object {
      "blendMode": "NORMAL",
      "color": Object {
        "b": 0.9607843160629272,
        "g": 0.9607843160629272,
        "r": 0.9607843160629272,
      },
      "opacity": 1,
      "type": "SOLID",
      "visible": true,
    },
  ],
  "children": Array [
    GeometryMixinStub {
      "_fills": Array [],
      "backgrounds": Array [
        Object {
          "color": Object {
            "b": 0,
            "g": 0,
            "r": 0,
          },
          "opacity": 0,
          "type": "SOLID",
        },
      ],
      "blendMode": "NORMAL",
      "children": Array [
        GeometryMixinStub {
          "_fills": Array [],
          "backgrounds": Array [],
          "blendMode": "NORMAL",
          "children": Array [
            GeometryMixinStub {
              "_fills": Array [
                Object {
                  "color": Object {
                    "b": 0.6666666666666666,
                    "g": 0.3333333333333333,
                    "r": 0.8666666666666667,
                  },
                  "opacity": 1,
                  "type": "SOLID",
                },
              ],
              "blendMode": "NORMAL",
              "config": Object {
                "isWithoutTimeout": false,
                "simulateErrors": true,
              },
              "constraints": Object {
                "horizontal": "MIN",
                "vertical": "MIN",
              },
              "cornerRadius": 0,
              "cornerSmoothing": 0,
              "effects": Array [],
              "exportSettings": Array [],
              "height": 100,
              "id": "1:2",
              "isMask": false,
              "layoutAlign": "INHERIT",
              "layoutGrow": 0,
              "locked": false,
              "opacity": 1,
              "parent": [Circular],
              "pluginData": Object {
                "isReactFigmaNode": "true",
                "reactStyle": "{\\"width\\":200,\\"height\\":100,\\"backgroundColor\\":\\"#dd55aa\\"}",
              },
              "strokeAlign": "INSIDE",
              "strokeCap": "NONE",
              "strokeWeight": 0,
              "strokes": Array [],
              "type": "RECTANGLE",
              "visible": true,
              "width": 200,
            },
            GeometryMixinStub {
              "_characters": "text",
              "_fills": Array [
                Object {
                  "color": Object {
                    "b": 1,
                    "g": 1,
                    "r": 1,
                  },
                  "opacity": 1,
                  "type": "SOLID",
                },
              ],
              "_fontName": Object {
                "family": "Roboto",
                "style": "Bold",
              },
              "_textAutoResize": "WIDTH_AND_HEIGHT",
              "autoRename": false,
              "blendMode": "NORMAL",
              "config": Object {
                "isWithoutTimeout": false,
                "simulateErrors": true,
              },
              "constraints": Object {
                "horizontal": "MIN",
                "vertical": "MIN",
              },
              "effects": Array [],
              "exportSettings": Array [],
              "fontSize": 50,
              "hyperlink": null,
              "id": "1:3",
              "isMask": false,
              "layoutAlign": "INHERIT",
              "layoutGrow": 0,
              "letterSpacing": Object {
                "unit": "PIXELS",
                "value": 0,
              },
              "lineHeight": Object {
                "unit": "AUTO",
              },
              "locked": false,
              "opacity": 1,
              "paragraphIndent": 0,
              "paragraphSpacing": 0,
              "parent": [Circular],
              "pluginData": Object {
                "isReactFigmaNode": "true",
                "reactStyle": "{\\"fontFamily\\":\\"Roboto\\",\\"fontSize\\":50,\\"fontWeight\\":\\"bold\\",\\"color\\":\\"#ffffff\\"}",
              },
              "strokeAlign": "INSIDE",
              "strokeCap": "NONE",
              "strokeWeight": 0,
              "strokes": Array [],
              "textAlignHorizontal": "LEFT",
              "textAlignVertical": "TOP",
              "textCase": "ORIGINAL",
              "textDecoration": "NONE",
              "type": "TEXT",
              "visible": true,
            },
          ],
          "clipsContent": false,
          "config": Object {
            "isWithoutTimeout": false,
            "simulateErrors": true,
          },
          "constraints": Object {
            "horizontal": "MIN",
            "vertical": "MIN",
          },
          "cornerRadius": 0,
          "cornerSmoothing": 0,
          "counterAxisAlignItems": "MIN",
          "counterAxisSizingMode": "AUTO",
          "description": "",
          "documentationLinks": Array [],
          "effects": Array [],
          "exportSettings": Array [],
          "guides": Array [],
          "id": "1:4",
          "isMask": false,
          "itemSpacing": 0,
          "layoutAlign": "INHERIT",
          "layoutGrids": Array [],
          "layoutGrow": 0,
          "layoutMode": "NONE",
          "locked": false,
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "parent": [Circular],
          "pluginData": Object {
            "isReactFigmaNode": "true",
            "reactStyle": "{}",
          },
          "primaryAxisAlignItems": "MIN",
          "primaryAxisSizingMode": "AUTO",
          "strokeAlign": "INSIDE",
          "strokeCap": "NONE",
          "strokeWeight": 0,
          "strokes": Array [],
          "type": "COMPONENT",
          "visible": true,
        },
      ],
      "clipsContent": false,
      "config": Object {
        "isWithoutTimeout": false,
        "simulateErrors": true,
      },
      "constraints": Object {
        "horizontal": "MIN",
        "vertical": "MIN",
      },
      "cornerRadius": 0,
      "cornerSmoothing": 0,
      "counterAxisAlignItems": "MIN",
      "counterAxisSizingMode": "AUTO",
      "effects": Array [],
      "exportSettings": Array [],
      "guides": Array [],
      "id": "1:5",
      "isMask": false,
      "itemSpacing": 0,
      "layoutAlign": "INHERIT",
      "layoutGrids": Array [],
      "layoutGrow": 0,
      "layoutMode": "NONE",
      "locked": false,
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
      "parent": [Circular],
      "pluginData": Object {
        "isReactFigmaNode": "true",
        "reactStyle": "{\\"backgroundColor\\":\\"transparent\\",\\"fontSize\\":50,\\"fontWeight\\":\\"bold\\",\\"fontFamily\\":\\"Roboto\\"}",
      },
      "primaryAxisAlignItems": "MIN",
      "primaryAxisSizingMode": "AUTO",
      "strokeAlign": "INSIDE",
      "strokeCap": "NONE",
      "strokeWeight": 0,
      "strokes": Array [],
      "type": "FRAME",
      "visible": true,
    },
  ],
  "config": Object {
    "isWithoutTimeout": false,
    "simulateErrors": true,
  },
  "exportSettings": Array [],
  "id": "1:1",
  "name": "New page",
  "parent": ChildrenMixinStub {
    "children": Array [
      ExportMixinStub {
        "children": Array [],
        "config": undefined,
        "id": "0:1",
        "parent": [Circular],
        "type": "PAGE",
      },
      [Circular],
    ],
    "config": undefined,
    "id": "0:0",
    "type": "DOCUMENT",
  },
  "pluginData": Object {
    "isReactFigmaNode": "true",
    "reactStyle": "{}",
  },
  "type": "PAGE",
}
`;

exports[`useInheritStyle styles propagating works with wrapping to component 1`] = `
ExportMixinStub {
  "_backgrounds": Array [
    Object {
      "blendMode": "NORMAL",
      "color": Object {
        "b": 0.9607843160629272,
        "g": 0.9607843160629272,
        "r": 0.9607843160629272,
      },
      "opacity": 1,
      "type": "SOLID",
      "visible": true,
    },
  ],
  "children": Array [
    GeometryMixinStub {
      "_fills": Array [],
      "backgrounds": Array [
        Object {
          "color": Object {
            "b": 0,
            "g": 0,
            "r": 0,
          },
          "opacity": 0,
          "type": "SOLID",
        },
      ],
      "blendMode": "NORMAL",
      "children": Array [
        GeometryMixinStub {
          "_fills": Array [],
          "backgrounds": Array [
            Object {
              "color": Object {
                "b": 0,
                "g": 0,
                "r": 0,
              },
              "opacity": 0,
              "type": "SOLID",
            },
          ],
          "blendMode": "NORMAL",
          "children": Array [
            GeometryMixinStub {
              "_fills": Array [
                Object {
                  "color": Object {
                    "b": 0.6666666666666666,
                    "g": 0.3333333333333333,
                    "r": 0.8666666666666667,
                  },
                  "opacity": 1,
                  "type": "SOLID",
                },
              ],
              "blendMode": "NORMAL",
              "config": Object {
                "isWithoutTimeout": false,
                "simulateErrors": true,
              },
              "constraints": Object {
                "horizontal": "MIN",
                "vertical": "MIN",
              },
              "cornerRadius": 0,
              "cornerSmoothing": 0,
              "effects": Array [],
              "exportSettings": Array [],
              "height": 100,
              "id": "1:2",
              "isMask": false,
              "layoutAlign": "INHERIT",
              "layoutGrow": 0,
              "locked": false,
              "opacity": 1,
              "parent": [Circular],
              "pluginData": Object {
                "isReactFigmaNode": "true",
                "reactStyle": "{\\"width\\":200,\\"height\\":100,\\"backgroundColor\\":\\"#dd55aa\\"}",
              },
              "strokeAlign": "INSIDE",
              "strokeCap": "NONE",
              "strokeWeight": 0,
              "strokes": Array [],
              "type": "RECTANGLE",
              "visible": true,
              "width": 200,
            },
            GeometryMixinStub {
              "_characters": "text",
              "_fills": Array [
                Object {
                  "color": Object {
                    "b": 1,
                    "g": 1,
                    "r": 1,
                  },
                  "opacity": 1,
                  "type": "SOLID",
                },
              ],
              "_fontName": Object {
                "family": "Roboto",
                "style": "Bold",
              },
              "_textAutoResize": "WIDTH_AND_HEIGHT",
              "autoRename": false,
              "blendMode": "NORMAL",
              "config": Object {
                "isWithoutTimeout": false,
                "simulateErrors": true,
              },
              "constraints": Object {
                "horizontal": "MIN",
                "vertical": "MIN",
              },
              "effects": Array [],
              "exportSettings": Array [],
              "fontSize": 48,
              "hyperlink": null,
              "id": "1:3",
              "isMask": false,
              "layoutAlign": "INHERIT",
              "layoutGrow": 0,
              "letterSpacing": Object {
                "unit": "PIXELS",
                "value": 0,
              },
              "lineHeight": Object {
                "unit": "AUTO",
              },
              "locked": false,
              "opacity": 1,
              "paragraphIndent": 0,
              "paragraphSpacing": 0,
              "parent": [Circular],
              "pluginData": Object {
                "isReactFigmaNode": "true",
                "reactStyle": "{\\"fontFamily\\":\\"Roboto\\",\\"fontSize\\":48,\\"fontWeight\\":\\"bold\\",\\"color\\":\\"#ffffff\\"}",
              },
              "strokeAlign": "INSIDE",
              "strokeCap": "NONE",
              "strokeWeight": 0,
              "strokes": Array [],
              "textAlignHorizontal": "LEFT",
              "textAlignVertical": "TOP",
              "textCase": "ORIGINAL",
              "textDecoration": "NONE",
              "type": "TEXT",
              "visible": true,
            },
          ],
          "clipsContent": false,
          "config": Object {
            "isWithoutTimeout": false,
            "simulateErrors": true,
          },
          "constraints": Object {
            "horizontal": "MIN",
            "vertical": "MIN",
          },
          "cornerRadius": 0,
          "cornerSmoothing": 0,
          "counterAxisAlignItems": "MIN",
          "counterAxisSizingMode": "AUTO",
          "effects": Array [],
          "exportSettings": Array [],
          "guides": Array [],
          "id": "1:4",
          "isMask": false,
          "itemSpacing": 0,
          "layoutAlign": "INHERIT",
          "layoutGrids": Array [],
          "layoutGrow": 0,
          "layoutMode": "NONE",
          "locked": false,
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "parent": [Circular],
          "pluginData": Object {
            "isReactFigmaNode": "true",
            "reactStyle": "{\\"backgroundColor\\":\\"transparent\\",\\"fontSize\\":48}",
          },
          "primaryAxisAlignItems": "MIN",
          "primaryAxisSizingMode": "AUTO",
          "strokeAlign": "INSIDE",
          "strokeCap": "NONE",
          "strokeWeight": 0,
          "strokes": Array [],
          "type": "FRAME",
          "visible": true,
        },
      ],
      "clipsContent": false,
      "config": Object {
        "isWithoutTimeout": false,
        "simulateErrors": true,
      },
      "constraints": Object {
        "horizontal": "MIN",
        "vertical": "MIN",
      },
      "cornerRadius": 0,
      "cornerSmoothing": 0,
      "counterAxisAlignItems": "MIN",
      "counterAxisSizingMode": "AUTO",
      "effects": Array [],
      "exportSettings": Array [],
      "guides": Array [],
      "id": "1:5",
      "isMask": false,
      "itemSpacing": 0,
      "layoutAlign": "INHERIT",
      "layoutGrids": Array [],
      "layoutGrow": 0,
      "layoutMode": "NONE",
      "locked": false,
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
      "parent": [Circular],
      "pluginData": Object {
        "isReactFigmaNode": "true",
        "reactStyle": "{\\"backgroundColor\\":\\"transparent\\",\\"fontSize\\":50,\\"fontWeight\\":\\"bold\\",\\"fontFamily\\":\\"Roboto\\"}",
      },
      "primaryAxisAlignItems": "MIN",
      "primaryAxisSizingMode": "AUTO",
      "strokeAlign": "INSIDE",
      "strokeCap": "NONE",
      "strokeWeight": 0,
      "strokes": Array [],
      "type": "FRAME",
      "visible": true,
    },
  ],
  "config": Object {
    "isWithoutTimeout": false,
    "simulateErrors": true,
  },
  "exportSettings": Array [],
  "id": "1:1",
  "name": "New page",
  "parent": ChildrenMixinStub {
    "children": Array [
      ExportMixinStub {
        "children": Array [],
        "config": undefined,
        "id": "0:1",
        "parent": [Circular],
        "type": "PAGE",
      },
      [Circular],
    ],
    "config": undefined,
    "id": "0:0",
    "type": "DOCUMENT",
  },
  "pluginData": Object {
    "isReactFigmaNode": "true",
    "reactStyle": "{}",
  },
  "type": "PAGE",
}
`;
